metadata-ingestion/src/datahub/ingestion/source/snowflake/snowflake_schema_gen.py

```diff
@@ -841,9 +847,11 @@ def gen_dataset_workunits(
             typeNames=(
                 [DatasetSubTypes.SNOWFLAKE_STREAM]
                 if isinstance(table, SnowflakeStream)
-                else [DatasetSubTypes.VIEW]
-                if isinstance(table, SnowflakeView)
-                else [DatasetSubTypes.TABLE]
+                else (
```

Comment:
maybe we should invert this e.g. SnowflakeStream and SnowflakeView etc have a `get_subtype()` method

Reply:
Done